### PR TITLE
Feature/prml 398 local mode should trigger preprocess and predict script

### PR DESCRIPTION
**Pull request by joker2411**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/ml_core/preprocess_and_predict.py [hunk 1]
```diff
@@ -257,4 +262,5 @@ def predict_scores_rs(df: pd.DataFrame) -> pd.Series:
         trainer=trainer,
     )
     logger.debug(f"Deleting additional local directory from infra mode")
```
Comment (dpatchigolla) [inline src/ml_core/preprocess_and_predict.py:264]: Shouldn't this message be inside the if statement?

File: src/processors/AWSProcessor.py [hunk 2]
```diff
@@ -111,12 +111,12 @@ def predict(
...
         )
 
-        logger.debug("Starting prediction on ec2")
+        logger.debug("Starting prediction on Rudderstack processing mode")
         ssm_client = boto3.client(service_name="ssm", region_name=s3_config["region"])
         commands = [
             f"cd {remote_dir}/rudderstack-profiles-classifier",
             f"pip install -r requirements.txt",
-            f"python3 -m src.ml_core.preprocess_and_predict.py --wh_creds '{json.dumps(creds)}' --s3_config '{json.dumps(s3_config)}' --json_output_filename {json_output_filename} --inputs '{json.dumps(inputs)}' --output_tablename {output_tablename} --merged_config '{json.dumps(merged_config)}' --prediction_task {prediction_task}",
+            f"python3 -m src.ml_core.preprocess_and_predict.py --wh_creds '{json.dumps(creds)}' --s3_config '{json.dumps(s3_config)}' --json_output_filename {json_output_filename} --inputs '{json.dumps(inputs)}' --output_tablename {output_tablename} --merged_config '{json.dumps(merged_config)}' --prediction_task {prediction_task} --mode {constants.RUDDERSTACK_MODE}",
```
Comment (dpatchigolla) [inline src/processors/AWSProcessor.py:119]: Is the mode not required for predict in K8SProcessor?

File: src/ml_core/preprocess_and_predict.py (hunk 1 above)
Comment (joker2411) [reply src/ml_core/preprocess_and_predict.py:264]: already moved it inside if in the 2nd PR.

File: src/processors/AWSProcessor.py (hunk 2 above)
Comment (joker2411) [reply src/processors/AWSProcessor.py:119]: K8sProcessor was already passing the mode argument. so, existing already.

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**